### PR TITLE
Send Request Backend Endpoint Consumption

### LINKED ISSUE #45: feat: Get member details from the backend
## Description
As a developer,
I need to connect to the backend,
so that I can send relation requests to the member.

## Mocks
[INSERT RELEVANT PNG FILE]

### Update [Required]
- [ ] Relation requests are posted to the backend.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
[INSERT NUMBER HERE] hours

### PR DESCRIPTION
**Pull request by yugantarjain**
### Description
This PR adds the backend endpoint consumption for send request. 

Additionally, also adds:
1. Activity indicator and text label for error messages
2. HomeModel and ProfileModel and parallel URL requests in home screen to load and store user info to get ID.

Fixes #45 

### Type of Change:
- Code
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Tested on Simulator and Postman.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

File: mentorship ios/Members/MemberDetail.swift
Comment (ApheleiaS) [inline mentorship ios/Members/MemberDetail.swift]: Add the title strings as keys?

File: mentorship ios/Members/SendRequest.swift
Comment (ApheleiaS) [inline mentorship ios/Members/SendRequest.swift]: timestamp is a misleading variable name

File: mentorship ios/Members/SupportingViews/MemberDetailCell.swift
Comment (ApheleiaS) [inline mentorship ios/Members/SupportingViews/MemberDetailCell.swift]: guard might be a better practice. Early return in case view is empty.

File: mentorship ios/Members/SupportingViews/SendRequestForm.swift
Comment (ApheleiaS) [inline mentorship ios/Members/SupportingViews/SendRequestForm.swift]: Explore using spacers for spacing?

File: mentorship ios/Members/SupportingViews/MemberDetailCell.swift
Comment (yugantarjain) [reply mentorship ios/Members/SupportingViews/MemberDetailCell.swift]: okay

File: mentorship ios/Members/SupportingViews/SendRequestForm.swift
Comment (yugantarjain) [reply mentorship ios/Members/SupportingViews/SendRequestForm.swift]: spacers act as a list cell and adds the separator line. We don't want that line.

File: mentorship ios/Members/SendRequest.swift
Comment (yugantarjain) [reply mentorship ios/Members/SendRequest.swift]: okay

**Review by vatsalkul — COMMENTED** [review]
If we are going with this PR, I have the following suggestion:



Please show an alertView rather than just a text if the request was sent successfully because it's look like an error. Like below:


After clicking on OK in alertView, we can dismiss the send request form.

**Comment by yugantarjain** [comment]
@vatsalkul yes, I have to do that.

**Comment by vatsalkul** [comment]
Here are the two point, just a little edit😅: 
1. In Models/MembersModel line 40, Change "Available to be a Mentor and Mentee both" to "Available to be a Mentor or Mentee" (suggested by ginny) and My suggestion: localise them also
2. In Members/SendRequest.swift line 30, localise “My Role” also (as suggested by akansha)

File: mentorship ios/Models/ProfileModel.swift
Comment (annabauza) [inline mentorship ios/Models/ProfileModel.swift]: can we remove this?

File: mentorship ios/Models/HomeModel.swift
Comment (annabauza) [inline mentorship ios/Models/HomeModel.swift]: Can we remove this?

File: mentorship ios/Members/SupportingViews/SendRequestForm.swift
Comment (annabauza) [inline mentorship ios/Members/SupportingViews/SendRequestForm.swift]: can we remove this?

File: mentorship ios/Members/SupportingViews/MemberDetailCell.swift
Comment (annabauza) [inline mentorship ios/Members/SupportingViews/MemberDetailCell.swift]: can we remove this?

File: mentorship ios/Constants/UserDefaultsConstants.swift
Comment (annabauza) [inline mentorship ios/Constants/UserDefaultsConstants.swift]: can we remove this?

**Review by annabauza — CHANGES_REQUESTED** [review]
please remove compyrights

File: mentorship ios/Members/SendRequest.swift
Comment (vatsalkul) [inline mentorship ios/Members/SendRequest.swift:63]: It's kind of unnecessary. Try optional (?) on sendRequestResponseData.message instead.As it is already initialised in MembersModel

File: mentorship ios/Models/ProfileModel.swift
Comment (vatsalkul) [reply mentorship ios/Models/ProfileModel.swift:6]: @sunjunkie Why this in happening even after adding custom file header?

**Review by vatsalkul — COMMENTED** [review]
> please remove compyrights

@yugantarjain  have already made the files before the merge of custom file header PR by @sunjunkie. That's why this happened. Please update the files header

File: mentorship ios/Members/SendRequest.swift
Comment (vatsalkul) [reply mentorship ios/Members/SendRequest.swift:63]: Or you can also use guard here

Comment (yugantarjain) [reply mentorship ios/Members/SendRequest.swift:63]: I don't think guard can be used here. Because then we have to return a view in the else closure. Using if statements is recommended by Apple.

Comment (vatsalkul) [reply mentorship ios/Members/SendRequest.swift:63]: I was just suggesting for this line: !(membersModel.sendRequestResponseData.message ?? ""). You are force unwrapping the value which always has a value i.e. ""
removing of ! would also do the job.

Comment (yugantarjain) [reply mentorship ios/Members/SendRequest.swift:63]: ! is not to force unwrap, but for the bool. To make sure that the message string isn't empty.

**Review by vatsalkul — APPROVED** [review]

File: mentorship ios/Members/SendRequest.swift
Comment (vatsalkul) [reply mentorship ios/Members/SendRequest.swift:63]: Oh! I think I was lost during the review. 😂

**Review by annabauza — APPROVED** [review]

**Pull request merged**
